src/map_extras.cpp

```diff
@@ -1116,87 +1116,94 @@ static bool mx_minefield( map &m, const tripoint &abs_sub )
 
     bool did_something = false;
 
-    if( bridge_at_north && bridge_at_center && road_at_south ) {
+    if( !bridgehead_at_center ) {
+        return false;
+    }
+
+    tinymap tm;
+    if( bridge_at_north && bridgehead_at_center && road_at_south ) {
+        tm.load( omt_to_sm_copy( abs_omt + point_south ), false );
```

Comment:
This looks strange. The function `mx_minefield` is given a map object and it *used* to create the minefield on that map. Now it creates the minefield on an arbitrary nearby map.

This means it will be created on an area that has been created earlier. For example: the player walks north, the terrain at (10, 10, 0) is created (as the reality bubble contains it now). The player can see that area in the distance (at least some parts of it). They see, there is no minefield. They move further, now (11, 10, 0) will be created. And that triggered placing a minefield, and that decided to place the minefield at (10, 10, 0), so it changes an area the player has already discovered!

Reply:
This stupid minefield map extra has driven me crazy.
* Road extras only appear on 1 in 75 roads, so if it's a road extra, it doesn't spawn frequently enough to reliably pass the minefield creation unit test because there are generally only 30-50 bridgeheads in the test area.
* The current map extra structure is too complicated to easily modify but takes up a full 24 tiles, which means I can't fit it on a bridgehead_ground map because of the ramp taking up space.
* slapping it on an adjacent map tile is ugly but it's the best solution I can think of, especially given that I don't care about minefields, I'm trying to get bridges and vehicle ramps to work right.

Reply:
It's rather... unpleasant to see you call my work stupid.

Reply:
I think the best solution is to modify it to fit on `bridgehead_ground`.  It's reasonable to punt that to a future PR, IMHO.

Reply:
Sorry, Night-Pryanik, I was just very frustrated with it.  It's not a bad thing at all, but I spent most of a week trying to get the minefield extra to work and the solution is admittedly not very satisfactory.